### PR TITLE
Feature/sign up

### PR DESCRIPTION
**Pull request by JuanfraM**
This PR contains: 
* Feature of sign UP

### CODE REVIEW

File: src/components/user/SignUpForm/styles.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  container: {
+    alignItems: 'center',
+    flex: 1,
+    top: 200,
+    width: 80
+  },
+  error: {
+    color: redColor
+  },
+  buttonContainer: {
```
Comment (felipegarcia92) [inline src/components/user/SignUpForm/styles.js]: CSS classes are written `with-hyphens`

File: src/containers/SignUpScreen/index.js [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+    <ImageBackground source={imageSource} style={styles.image}>
+      <Text style={styles.textCenter}>TARGET MVD</Text>
+    </ImageBackground>
+    <SignUpForm onSubmit={user => signUp(user.toJS())} navigator={navigator} />
   </View>
 );
 
+const { func } = PropTypes;
+
+SignUpScreen.propTypes = {
```
Comment (felipegarcia92) [inline src/containers/SignUpScreen/index.js:22]: Don't you need to add `navigator` to PropTypes?

If you don't have the eslinter configured into your text editor to warn you about these, I recommend you configure it.

**Review by felipegarcia92 — APPROVED** [review]
👍

File: src/actions/userActions.js [hunk 3]
```diff
@@ -25,6 +25,19 @@ export const login = user =>
     }
   };
 
+export const signUp = user =>
+  async (dispatch) => {
+    try {
+      const response = await userApi.signUp({ user });
```
Comment (agch-dev) [inline src/actions/userActions.js:31]: here you can just do `const { user } = ...`

File: src/actions/userActions.js [hunk 4]
```diff
@@ -25,6 +25,19 @@ export const login = user =>
...
+      await sessionService.saveUser(response.user);
+      dispatch(loginSuccess());
+    } catch (err) {
```
Comment (agch-dev) [inline src/actions/userActions.js:34]: destructure `error`

File: src/components/common/Input/index.js [hunk 5]
```diff
@@ -3,12 +3,21 @@ import PropTypes from 'prop-types';
 import { View, TextInput, Text, ViewPropTypes } from 'react-native';
 import styles from './styles';
 
-const Input = ({ input: { onChange, ...restInput }, inputContainer, password = false, label, meta: { touched, error } }) => (
+const Input = (
+  { input: { onChange, ...restInput },
```
Comment (agch-dev) [inline src/components/common/Input/index.js]: wrong indentation

File: src/actions/userActions.js (hunk 3 above)
Comment (JuanfraM) [reply src/actions/userActions.js:31]: I couldn't  because it intends to rename the existing user var

File: src/actions/userActions.js [hunk 6]
```diff
@@ -18,9 +18,9 @@ export const login = user =>
       const response = await userApi.login({ user });
       await sessionService.saveUser(response.user);
       dispatch(loginSuccess());
-    } catch (err) {
+    } catch ({ error }) {
       throw new SubmissionError({
-        _error: err.error,
+        error,
```
Comment (agch-dev) [inline src/actions/userActions.js]: you sill need to do `_error: error` to set the global error in redux form

File: src/actions/userActions.js (hunk 3 above)
Comment (agch-dev) [reply src/actions/userActions.js:31]: oh ok, you can always do sth like `const { user: newUser } = ...` but in this case I think it's not worth it. That is useful when you have long paths to variables that are named the same.

File: src/actions/userActions.js [hunk 7]
```diff
@@ -31,9 +31,9 @@ export const signUp = user =>
       const response = await userApi.signUp({ user });
       await sessionService.saveUser(response.user);
       dispatch(loginSuccess());
-    } catch (err) {
+    } catch ({ error }) {
       throw new SubmissionError({
-        _error: err.error,
+        error,
```
Comment (agch-dev) [inline src/actions/userActions.js]: same here

File: src/components/user/SignUpForm/styles.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  blackColor,
+  whiteColor
+} from '../../../constants/styleConstants';
+
+const styles = StyleSheet.create({
+  container: {
+    alignItems: 'center',
+    flex: 1,
+    top: 200,
+    width: 80
```
Comment (juanlacu) [inline src/components/user/SignUpForm/styles.js]: You shouldn't use fixed values on weight or height. Try to use percentages instead.

https://medium.com/react-native-training/build-responsive-react-native-views-for-any-device-and-support-orientation-change-1c8beba5bc23

**Review by juanlacu — COMMENTED** [review]
Good one! 👌

File: src/components/user/SignUpForm/styles.js (hunk 8 above)
Comment (JuanfraM) [reply src/components/user/SignUpForm/styles.js]: Thanks 👌

**Review by juanlacu — APPROVED** [review]

**Review by agch-dev — APPROVED** [review]

**Pull request merged**